geode-protobuf/src/test/java/org/apache/geode/protocol/acceptance/LocatorConnectionDUnitTest.java

```diff
@@ -51,78 +58,234 @@
   public void testInvalidOperationReturnsFailure()
       throws IOException, InvalidProtocolMessageException {
-    ClientProtocol.Request.Builder protobufRequestBuilder =
-        ProtobufUtilities.createProtobufRequestBuilder();
-    ClientProtocol.Message getAvailableServersRequestMessage =
-        ProtobufUtilities.createProtobufMessage(ProtobufUtilities.createMessageHeader(1233445),
-            protobufRequestBuilder
-                .setGetRegionNamesRequest(ProtobufRequestUtilities.createGetRegionNamesRequest())
-                .build());
+    IgnoredException ignoredInvalidExecutionContext =
+        IgnoredException.addIgnoredException("Invalid execution context");
+    try (Socket socket = createSocket()) {
+
+      ClientProtocol.Request.Builder protobufRequestBuilder =
+          ProtobufUtilities.createProtobufRequestBuilder();
+      ClientProtocol.Message getRegionNamesRequestMessage =
+          ProtobufUtilities.createProtobufMessage(ProtobufUtilities.createMessageHeader(1233445),
+              protobufRequestBuilder
+                  .setGetRegionNamesRequest(ProtobufRequestUtilities.createGetRegionNamesRequest())
+                  .build());
+
+      long messagesReceived = getMessagesReceived();
+      long messagesSent = getMessagesSent();
+      int clientConnectionStarts = getClientConnectionStarts();
+      int clientConnectionTerminations = getClientConnectionTerminations();
+
+      ProtobufProtocolSerializer protobufProtocolSerializer = new ProtobufProtocolSerializer();
+      protobufProtocolSerializer.serialize(getRegionNamesRequestMessage, socket.getOutputStream());
+
+      ClientProtocol.Message getAvailableServersResponseMessage =
+          protobufProtocolSerializer.deserialize(socket.getInputStream());
+      assertEquals(1233445,
+          getAvailableServersResponseMessage.getMessageHeader().getCorrelationId());
+      assertEquals(ClientProtocol.Message.MessageTypeCase.RESPONSE,
+          getAvailableServersResponseMessage.getMessageTypeCase());
+      ClientProtocol.Response messageResponse = getAvailableServersResponseMessage.getResponse();
+      assertEquals(ClientProtocol.Response.ResponseAPICase.ERRORRESPONSE,
+          messageResponse.getResponseAPICase());
+      assertEquals(ProtocolErrorCode.UNSUPPORTED_OPERATION.codeValue,
+          messageResponse.getErrorResponse().getError().getErrorCode());
+
+      Host.getLocator().invoke(() -> {
+        InternalDistributedSystem distributedSystem =
+            (InternalDistributedSystem) Locator.getLocator().getDistributedSystem();
+
+        Statistics[] protobufServerStats = distributedSystem
+            .findStatisticsByType(distributedSystem.findType("ProtobufServerStats"));
+        assertEquals(1, protobufServerStats.length);
+        Statistics statistics = protobufServerStats[0];
+        assertEquals(0, statistics.get("currentClientConnections"));
+        assertEquals(messagesReceived + 1, statistics.get("messagesReceived"));
```

Comment:
Good idea as it can definitely be refactored. My hesitancy was due to getting a nice stack trace in the event of failures but that looks like it works reasonably.